scripts/run_linter.lean

```diff
@@ -0,0 +1,167 @@
+      let defaultLinters ← getChecksNew (slow := true) (useAlso := newNames)
+      let linters := if updateOnlyRemove then
+        -- Determine all unique linters which are mentioned in the `nolints` file.
+        let all := (nolints.map fun nol ↦ nol.1).qsort (·.toString < ·.toString)
+        let uniqueLinters := (all.toList).pwFilter (·.toString ≠ ·.toString)
+        defaultLinters.filter fun lint ↦ uniqueLinters.contains lint.name
+      else
+        if let some only := only then
+          if let some extra := add then
+            defaultLinters.filter fun lint ↦
+              extra.contains lint.name.toString || only.contains lint.name.toString
+          else
+            defaultLinters.filter fun lint ↦ only.contains lint.name.toString
+        else if let some exclude := exclude then
+          if let some extra := add then
+            -- add takes priority over --exclude: a lint must be excluded and *not* added
+            defaultLinters.filter fun lint ↦
+              !exclude.contains lint.name.toString && !extra.contains lint.name.toString
+          else
+            defaultLinters.filter (fun lint ↦ !exclude.contains lint.name.toString)
+        else defaultLinters
+      if print then
+        IO.println s!"The following {linters.size} linter(s) were configured to run: \
+          {linters.map fun l ↦ l.name}"
+        IO.Process.exit 0
+
+      let results ← lintCore decls linters
+      if update then
+        writeJsonFile (α := NoLints) nolintsFile <|
+          .qsort (lt := fun (a, b) (c, d) ↦ a.lt c || (a == c && b.lt d)) <|
+          .flatten <| results.map fun (linter, decls) ↦
+          decls.fold (fun res decl _ ↦ res.push (linter.name, decl)) #[]
+      let results := results.map fun (linter, decls) ↦
+        .mk linter <| nolints.foldl (init := decls) fun decls (linter', decl') ↦
+          if linter.name == linter' then decls.erase decl' else decls
+      let failed := results.any (!·.2.isEmpty)
+      if failed then
+        let fmtResults ←
+          formatLinterResults results decls (groupByFilename := true) (useErrorFormat := true)
+            s!"in {module}" (runSlowLinters := true) .medium linters.size
+        IO.print (← fmtResults.toString)
+        IO.Process.exit 1
+      else
+        IO.println "-- Linting passed."
+  return 0
+
+/-- Setting up command line options and help text for `lake exe myRunLinter`. -/
+unsafe def runLinter : Cmd := `[Cli|
+  lint_style VIA runLinterCli; ["0.0.1"]
+  "Runs the linters on all declarations in the given module (or `Mathlib` by default)."
+
+  FLAGS:
+    only_linters : Array String;  "Only run these named linters"
+    exclude_linters : Array String; "Do not run these named linters"
+    add_linters : Array String; "Run these linters *in addition* to the default set\n\
+      Takes priority over the --only_linters or --exclude_linters flags."
+    print_linters; "Print the list of all discovered/configured linters and exit"
```

Comment:
Remove `_linters` from these options. I would recommend using `--only`, `--ignore`, `--select`, and `--list` for these options (`include` and `exclude` are usually for files to lint or avoid linting). The priority should be on the last option provided. So `--select foo --ignore foo` would ignore, but `--ignore foo --select foo` would run `foo`.

Reply:
Thanks for the feedback! I like your renaming suggestions, I'll do that in a moment.

I'm not sure regarding the order of options matter (my current instinct is to complain about contradictory information). Do you have any examples in mind which are order-sensitive --- or cases where specifying contradictory information is actually useful?

Reply:
It is useful for when you create a shell alias `alias mylint="lake exe run_linter --only foo"` and then using the alias `mylint --select bar` will work as expected.